In [1]:
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

# Read the data set into python

In [2]:
df=pd.read_excel('Dataset_w6.xlsx')
df

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0              652864.0                         35530.0   
1               28748.0                          2930.0   
2             2381741.0                         41318.0   
3             1246700.0                         29784.0   
4                 442.0                           102.0   
..                  ...                             ...   
191            912050.0                         31977.0   
192            330967.0                         95541.0   
193            527968.0                         28250.0   
194            752612.0                         17094.0   
195            390757.0                         16530.0   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                           125.0   
1                          3984.2                           134.0   
2                          4154.1                           160.0   
3                          4714.1                           175.0   
4                         14764.5                            88.0   
..                            ...                             ...   
191                       11068.9                           118.0   
192                        2067.9                           136.0   
193                        1106.4                           136.0   
194                        1311.1                           179.0   
195                         890.4                            99.0   

     International trade: Imports (million US$)  
0                                        3568.0  
1                                        4669.0  
2                                       47091.0  
3                                        8790.0  
4                                         491.0  
..                                          ...  
191                                     16324.0  
192                                    174111.0  
193                                      6861.0  
194                                      7442.0  
195                                      5212.0  

[196 rows x 7 columns]

# 1. Find out how many countries are there in the dataset.

In [3]:
count=df['country'].count()
count

196

The dataset have 196 countries.

# 2. Find out how many countries with respect to the region are available in the dataset

In [4]:
df["Region"].value_counts()

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthAmerica         12
SouthernEurope       12
South-easternAsia    11
EasternEurope        10
NorthernEurope       10
SouthernAsia          9
MiddleAfrica          9
CentralAmerica        8
WesternEurope         7
EasternAsia           7
SouthernAfrica        5
Polynesia             5
NorthernAfrica        5
Melanesia             5
CentralAsia           5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

In [5]:
len(df["Region"].value_counts())

22

# 3. On average which region has the highest  GDP per capita (current US$)

In [6]:
df2=df[['Region','GDP per capita (current US$)']].groupby('Region').mean()
df2=df2.sort_values('GDP per capita (current US$)',ascending=False)
df2

GDP per capita (current US$)
Region                                         
NorthernAmerica                    57659.000000
WesternEurope                      55387.128571
Oceania                            44823.250000
NorthernEurope                     42098.220000
EasternAsia                        27967.614286
WesternAsia                        17013.938889
Caribbean                          15135.562500
SouthernEurope                     14462.741667
South-easternAsia                  10299.027273
EasternEurope                       9292.490000
Melanesia                           9261.960000
Polynesia                           8686.520000
SouthAmerica                        8499.775000
CentralAmerica                      6348.137500
Micronesia                          5008.300000
CentralAsia                         4329.880000
SouthernAfrica                      4191.900000
NorthernAfrica                      3934.960000
MiddleAfrica                        3928.833333
SouthernAsia                        2968.400000
EasternAfrica                       2088.852941
WesternAfrica                       1013.140000

# 3. On average which region has the highest International trade: Imports (million US$)

In [7]:
df4=df[['Region','International trade: Imports (million US$)']].groupby('Region').mean()
df4=df4.sort_values('International trade: Imports (million US$)',ascending=False)
df4

International trade: Imports (million US$)
Region                                                       
NorthernAmerica                                 663552.250000
EasternAsia                                     451995.000000
WesternEurope                                   407498.428571
NorthernEurope                                  113407.200000
Oceania                                         112914.500000
South-easternAsia                                98324.636364
EasternEurope                                    85243.900000
SouthernEurope                                   76666.666667
SouthernAsia                                     58199.333333
CentralAmerica                                   57031.500000
WesternAsia                                      46025.555556
NorthernAfrica                                   35034.400000
SouthAmerica                                     32885.666667
SouthernAfrica                                   18005.600000
CentralAsia                                      12063.400000
WesternAfrica                                     4898.333333
EasternAfrica                                     4726.176471
MiddleAfrica                                      4381.222222
Caribbean                                         3057.312500
Melanesia                                         2896.400000
Polynesia                                          432.000000
Micronesia                                          99.000000

# 4. Find out which region has more consistent with respect to GDP per capita (current US$)

In [8]:
df2=df[['Region','GDP per capita (current US$)']].groupby('Region').std()
df2=df2.sort_values('GDP per capita (current US$)',ascending=True)
df2

GDP per capita (current US$)
Region                                         
WesternAfrica                        815.095399
NorthernAfrica                       974.706575
SouthernAfrica                      2177.551120
SouthernAsia                        2855.708179
EasternAfrica                       3716.131770
CentralAsia                         4152.215490
CentralAmerica                      4180.440883
SouthAmerica                        4273.707394
Micronesia                          4820.669694
MiddleAfrica                        5274.846281
EasternEurope                       5415.488739
Polynesia                           6956.250286
Oceania                             9233.329638
SouthernEurope                      9541.703216
Melanesia                          13855.842686
Caribbean                          14579.371326
South-easternAsia                  16311.810656
WesternAsia                        18812.151459
NorthernEurope                     20671.030185
WesternEurope                      24770.467728
NorthernAmerica                    25749.619445
EasternAsia                        27485.120923

In [9]:
df4=df[['Region','International trade: Imports (million US$)']].groupby('Region').mean()
df4=df4.sort_values('International trade: Imports (million US$)',ascending=False)
df4

International trade: Imports (million US$)
Region                                                       
NorthernAmerica                                 663552.250000
EasternAsia                                     451995.000000
WesternEurope                                   407498.428571
NorthernEurope                                  113407.200000
Oceania                                         112914.500000
South-easternAsia                                98324.636364
EasternEurope                                    85243.900000
SouthernEurope                                   76666.666667
SouthernAsia                                     58199.333333
CentralAmerica                                   57031.500000
WesternAsia                                      46025.555556
NorthernAfrica                                   35034.400000
SouthAmerica                                     32885.666667
SouthernAfrica                                   18005.600000
CentralAsia                                      12063.400000
WesternAfrica                                     4898.333333
EasternAfrica                                     4726.176471
MiddleAfrica                                      4381.222222
Caribbean                                         3057.312500
Melanesia                                         2896.400000
Polynesia                                          432.000000
Micronesia                                          99.000000

In [19]:
cv= lambda x : np.std(x, ddof=1) / np.mean(x)*100
cv('Region')

TypeError: cannot perform reduce with flexible type